AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m

```diff
@@ -20,32 +20,44 @@
 static NSString *const kMSStorageReadOnlyDbTokenKey = @"MSStorageReadOnlyDbToken";
 static NSString *const kMSStorageUserDbTokenKey = @"MSStorageUserDbToken";
 
+/**
+ * The API paths for cosmosDb token.
+ */
+static NSString *const kMSGetTokenPath = @"/data/tokens";
+
 @implementation MSTokenExchange : NSObject
 
-+ (void)performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)httpClient
++ (void)performDbTokenAsyncOperationWithHttpClient:(id<MSHttpClientProtocol>)httpClient
+                                  tokenExchangeUrl:(NSURL *)tokenExchangeUrl
+                                         appSecret:(NSString *)appSecret
                                          partition:(NSString *)partition
-                                 completionHandler:(MSGetTokenAsyncCompletionHandler _Nonnull)completionHandler {
+                                 completionHandler:(MSGetTokenAsyncCompletionHandler)completionHandler {
 
   // Get the cached token if it is saved.
   MSTokenResult *cachedToken = [MSTokenExchange retrieveCachedToken:partition];
+  NSURL *sendUrl = [tokenExchangeUrl URLByAppendingPathComponent:kMSGetTokenPath];
 
   // Get a fresh token from the token exchange service if the token is not cached or has expired.
   if (!cachedToken) {
 
-    // Payload.
+    // Serialize payload.
     NSError *jsonError;
     NSData *payloadData = [NSJSONSerialization dataWithJSONObject:@{kMSPartitions : @[ partition ]} options:0 error:&jsonError];
 
+    // Call token exchange service.
+    NSMutableDictionary *headers = [NSMutableDictionary new];
+    headers[kMSHeaderContentTypeKey] = kMSAppCenterContentType;
```

Comment:
Not part of the change but could it be `kMSJsonContentType` instead?

Reply:
Yes but out of scope of this PR.